packages/react-devtools-shared/src/backend/fiber/renderer.js

```diff
@@ -3868,7 +4065,9 @@ export function attach(
         previouslyReconciledSibling = stashedPrevious;
         remainingReconcilingChildren = stashedRemaining;
         reconcilingParentSuspenseNode = stashedSuspenseParent;
-        previouslyReconciledSiblingSuspenseNode = stashedSuspensePrevious;
+        if (fiberInstance.suspenseNode !== null) {
+          previouslyReconciledSiblingSuspenseNode = stashedSuspensePrevious;
+        }
```

Comment:
This is required so that siblings stay connected if we're reordering nodes that contain Suspense e.g. in 
```
<Suspense>
  <Suspense name="a" />
  <div> 
    <Suspense name="b" />
  </div>
  <div> 
    <Suspense name="c" />
  </div>
  <Suspense name="d" />
```

reordering the containing divs meant the renderer constructed `a -> d`. I added an example to the shell that illustrates the problem.

I probably missed some more cases where we're restoring too early.